guiEditor/src/components/propertyTab/propertyGrids/gui/commonControlPropertyGridComponent.tsx

```diff
@@ -70,15 +70,14 @@ export class CommonControlPropertyGridComponent extends React.Component<ICommonC
         if (value.charAt(value.length - 1) === '%') {
             percentage = true;
         }
-        else if(value.charAt(value.length - 1) === 'x' && value.charAt(value.length - 2) === 'p') {
+        else if (value.charAt(value.length - 1) === 'x' && value.charAt(value.length - 2) === 'p') {
             percentage = false;
         }
-        let newValue = value.split('').filter(function (item) {
-            return (!isNaN(parseInt(item)));
-        }).join('');
 
+        let newValue = value.match(/([\d\.\,]+)/g)?.toString();

```

Comment:
have you tested this?

Reply:
ya I've been teasing this. unless I missed something?

Reply:
just not sure about the toString call on an optional variable :-) Or why it is needed? Check if the matches array has a value and use it instead of calling toString on an array. Because:

``` javascript
"123.23px43234".match(/([\d\.\,]+)/g).toString() === '123.23,43234';
// but
"123.23px43234".match(/([\d\.\,]+)/g) === ['123.23', '43234'];
```


If it is optional (i.e. `?` at the end) it means it can be undefined. so newValue will be undefined, which means the next line will fail.

Reply:
how's this? :)

Reply:
works ;-)